src/mdns/mdns_mdnssd.cpp

```diff
@@ -415,9 +462,133 @@ otbrError PublisherMDnsSd::PublishService(uint16_t aPort, const char *aName, con
     return ret;
 }
 
-Publisher *Publisher::Create(int aFamily, const char *aHost, const char *aDomain, StateHandler aHandler, void *aContext)
+otbrError PublisherMDnsSd::UnpublishService(const char *aName, const char *aType)
+{
+    DiscardService(aName, aType);
+    return OTBR_ERROR_NONE;
+}
+
+otbrError PublisherMDnsSd::PublishHost(const char *aName, const uint8_t *aAddress, uint8_t aAddressLength)
```

Comment:
Add check for the aAddressLength to be a valid IPv6 address or pass ip6_addr_t instead?